cortex/secondary/call_duration.py

```diff
@@ -27,15 +29,28 @@ def call_duration(incoming=True, **kwargs):
             timestamp (int): The beginning of the window (same as kwargs['start']).
             value (float): The time spent in a call.
     """
-    incoming_dict = {True: 1, False: 2}
-    label = incoming_dict[incoming]
-    log.info('Loading raw calls data...')
-    _calls = calls(**kwargs)['data']
-    log.info('Computing call duration...')
-    _call_duration = np.sum([call['call_duration'] for call in _calls
-                             if call['call_type'] == label])
+    log.info('Computing call duration ...')
+
+    _calls = telephony(**kwargs)['data']
+
+    call_duration = np.sum(call['duration'] for call in _calls)
+
+    if incoming == "all":
+        call_duration = np.sum(call['duration'] for call in _calls
+                               if (call['type'] == "incoming"
+                                   or call['type'] == "outgoing"))
+
+    if incoming == "incoming":
+        call_duration = np.sum(call['duration'] for call in _calls
+                               if call['type'] == "incoming")
+
+    if incoming == "outgoing":
+        call_duration = np.sum(call['duration'] for call in _calls
+                               if call['type'] == "outgoing")
+
     # if you have no call duration, this means you have no call data
-    # over the period, should return None
-    if _call_duration == 0:
-        _call_duration = None
-    return {'timestamp': kwargs['start'], 'value': _call_duration}
+
+    if call_duration == 0:
+        call_duration = None
```

Comment:
None should only be returned if there's *no* call data at all (i.e. if there are outgoing calls but no incoming calls it's fine to return 0 for duration of incoming)